source/common/formatter/stream_info_formatter.cc

```diff
@@ -313,6 +313,165 @@ FilterStateFormatter::formatValue(const StreamInfo::StreamInfo& stream_info) con
+        {FirstDownstreamRxByteReceived,
+         [](const StreamInfo::StreamInfo& stream_info) -> absl::optional<MonotonicTime> {
+           return stream_info.startTimeMonotonic();
+         }},
+        {LastDownstreamRxByteReceived,
+         [](const StreamInfo::StreamInfo& stream_info) -> absl::optional<MonotonicTime> {
+           const auto downstream_timing = stream_info.downstreamTiming();
+           if (downstream_timing.has_value()) {
+             return downstream_timing->lastDownstreamRxByteReceived();
+           }
+           return {};
+         }},
+        {FirstUpstreamTxByteSent,
+         [](const StreamInfo::StreamInfo& stream_info) -> absl::optional<MonotonicTime> {
+           const auto upstream_info = stream_info.upstreamInfo();
+           if (upstream_info.has_value()) {
+             return upstream_info->upstreamTiming().first_upstream_tx_byte_sent_;
+           }
+           return {};
+         }},
+        {LastUpstreamTxByteSent,
+         [](const StreamInfo::StreamInfo& stream_info) -> absl::optional<MonotonicTime> {
+           const auto upstream_info = stream_info.upstreamInfo();
+           if (upstream_info.has_value()) {
+             return upstream_info->upstreamTiming().last_upstream_tx_byte_sent_;
+           }
+           return {};
+         }},
+        {FirstUpstreamRxByteReceived,
+         [](const StreamInfo::StreamInfo& stream_info) -> absl::optional<MonotonicTime> {
+           const auto upstream_info = stream_info.upstreamInfo();
+           if (upstream_info.has_value()) {
+             return upstream_info->upstreamTiming().first_upstream_rx_byte_received_;
+           }
+           return {};
+         }},
+        {LastUpstreamRxByteReceived,
+         [](const StreamInfo::StreamInfo& stream_info) -> absl::optional<MonotonicTime> {
+           const auto upstream_info = stream_info.upstreamInfo();
+           if (upstream_info.has_value()) {
+             return upstream_info->upstreamTiming().last_upstream_rx_byte_received_;
+           }
+           return {};
+         }},
+        {FirstDownstreamTxByteSent,
+         [](const StreamInfo::StreamInfo& stream_info) -> absl::optional<MonotonicTime> {
+           const auto downstream_timing = stream_info.downstreamTiming();
+           if (downstream_timing.has_value()) {
+             return downstream_timing->firstDownstreamTxByteSent();
+           }
+           return {};
+         }},
+        {LastDownstreamTxByteSent,
+         [](const StreamInfo::StreamInfo& stream_info) -> absl::optional<MonotonicTime> {
+           const auto downstream_timing = stream_info.downstreamTiming();
+           if (downstream_timing.has_value()) {
+             return downstream_timing->lastDownstreamTxByteSent();
+           }
+           return {};
+         }},
+    };
+
+CommonDurationFormatter::TimePointGetter
+CommonDurationFormatter::getTimePointGetterByName(absl::string_view name) {
+  auto it = KnownTimePointGetters.find(name);
```

Comment:
I think either we should be case-insensitive here or explicitly document that these strings are case-sensitive in `usage.rst`. And that should apply to the precision specifiers as well.

Reply:
I will document that all these token are case-sensitive. Because the `Dynamic value` (used in the downstream timing by setValue or getValue) is case-sensitive. It would be better to keep align with it.